sklearn/multioutput.py

```diff
@@ -150,3 +195,74 @@ def score(self, X, y, sample_weight=None):
         from .metrics import r2_score
         return r2_score(y, self.predict(X), sample_weight=sample_weight,
                         multioutput='uniform_average')
+
+
+class MultiOutputClassifier(MultiOutputEstimator, ClassifierMixin):
+    """Multi target classification
+
+    This strategy consits of fitting one classifier per target. This is a
+    simple strategy for extending classifiers that do not natively support
+    multi-target classification
+
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit`,`score` and `predict_proba`.
+    n_jobs : int, optional, default=1
+        The number of jobs to use for the computation. If -1 all CPUs are used.
+        If 1 is given, no parallel computing code is used at all, which is
+        useful for debugging. For n_jobs below -1, (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        The number of jobs to use for the computation.
+        It does each target variable in y in parallel.
+    Attributes
+    ----------
+    estimators_ : list of `n_output` estimators
+        Estimators used for predictions.
+    """
+
+    def __init__(self, estimator, n_jobs=1):
+        super(MultiOutputClassifier, self).__init__(estimator, n_jobs)
+
+    def predict_proba(self, X):
+        """Probability estimates. This returns prediction probabilites
+        for each class for each label.
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        Returns
+        -------
+        T : (sparse) array-like, shape = [n_samples, n_classes, n_outputs]
+            The class probabilities of the samples for each of the outputs
+        """
+        check_is_fitted(self, 'estimators_')
+
+        results = np.dstack([estimator.predict_proba(X) for estimator in
+                            self.estimators_])
+        return results
+        
+    def score(self, X, y):
+        """"Returns the mean accuracy on the given test data and labels.
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        y : array-like, shape = [n_samples, n_outputs]
+        Returns
+        -------
+        scores : float
+            accuracy_score of self.predict(X) versus y
+        """
+        check_is_fitted(self, 'estimators_')
+        n_outputs_ = len(self.estimators_)
+        if y.ndim == 1:
+            raise ValueError("y must have at least two dimensions for "
+                             "multi target classification but has only one")
+        if y.shape[1] != n_outputs_:
+            raise ValueError("The number of outputs of Y for fit {0} and"
+                             " score {1} should be same".
+                             format(n_outputs_, y.shape[1]))
+        y_pred = sp.csr_matrix(self.predict(X))
```

Comment:
Thanks a lot. It is elegant ! I tried to come up on my own but later decided to follow the same idea as multi label.